File: eval/solutions/go/02-control-flow/main_test.go
package main

import "testing"

func TestBatteryHealth(t *testing.T) {
	tests := []struct {
		level int
		want  string
	}{
		{0, "critical"},
		{20, "critical"},
		{21, "low"},
		{50, "low"},
		{51, "healthy"},
		{99, "healthy"},
		{100, "full"},
	}
	for _, tc := range tests {
		got := BatteryHealth(tc.level)
		if got != tc.want {
			t.Errorf("BatteryHealth(%d) = %q, want %q", tc.level, got, tc.want)
		}
	}
}

File: rust/02-control-flow/tests/main_test.rs
use lesson_02_control_flow::battery_health;

#[test]
fn test_critical() {
    assert_eq!(battery_health(0), "critical");
    assert_eq!(battery_health(20), "critical");
}

#[test]
fn test_low() {
    assert_eq!(battery_health(21), "low");
    assert_eq!(battery_health(50), "low");
}

#[test]
fn test_healthy() {
    assert_eq!(battery_health(51), "healthy");
    assert_eq!(battery_health(99), "healthy");
}

#[test]
fn test_full() {
    assert_eq!(battery_health(100), "full");
}
